digital_land/commands.py

```diff
@@ -531,6 +533,232 @@ def collection_add_source(entry, collection, endpoint_url, collection_dir):
+                    except ValueError:
+                        raise ValueError(
+                            f"start-date {given_start_date} must be format YYYY-MM-DD"
+                        )
+
+                    if given_start_date > datetime.today().date():
+                        raise ValueError(
+                            f"The start_date {given_start_date} cannot be in the future"
+                        )
+                except ValueError:
+                    raise
+                except Exception as e:
+                    raise ValueError(f"An unexpected error occured: {str(e)}")
+
+            # validate organisation
+            if (
+                row["organisation"] == ""
+                or row["organisation"] not in valid_organisations
+            ):
+                if row["organisation"] == "":
+                    raise ValueError("The organisation must not be blank")
+                if row["organisation"] not in valid_organisations:
+                    raise ValueError(
+                        f"The given organisation '{row['organisation']}' is not in our valid organisations"
+                    )
+
+            # validate pipeline(s) - do they exist and are they in the collection
+            pipelines = row["pipelines"].split(";")
+            dataset_df = pd.read_csv(os.path.join(specification_dir, "dataset.csv"))
+            for pipeline in pipelines:
+                pipeline_row = dataset_df[dataset_df["dataset"] == pipeline]
+                if len(pipeline_row) == 0:
+                    raise ValueError(
+                        f"'{pipeline}' is not a valid dataset in the specification"
+                    )
+                if pipeline_row["collection"].values[0] != collection_name:
+                    raise ValueError(
+                        f"'{pipeline}' does not belong to provided collection {collection_name}"
+                    )
+
+    # VALIDATION DONE, NOW ADD TO COLLECTION
+    print("======================================================================")
+    print("Endpoint and source details")
+    print("======================================================================")
+    print("Endpoint URL: ", row["endpoint-url"])
+    print("Endpoint Hash:", hash_value(row["endpoint-url"]))
+    print("Documentation URL: ", row["documentation-url"])
+    print()
+
+    # if endpoint already exists, it will indicate it and quit function here
+    if collection.add_source_endpoint(row):
+        endpoint = {
+            "endpoint-url": row["endpoint-url"],
+            "endpoint": hash_value(row["endpoint-url"]),
+            "end-date": row.get("end-date", ""),
+            "plugin": row.get("plugin"),
+            "licence": row["licence"],
+        }
+        endpoints.append(endpoint)
+    # if successfully added we can now attempt to fetch from endpoint
```

Comment:
I think this part doesn't function as initially thought. It will print out that the endpoint already exists, but it doesn't actually end the function (the user prompt is still being asked)